### ISSUE TITLE
Semicolon in CIFS share password breaks mount

### ISSUE THREAD

**Issue by maleadt** [RuneAudio/RuneUI#44]
If I mount a CIFS share using the WebUI, and the password contains a semicolon mounting fails. The webpage displayed an error by `sh` indicating the semicolon spawned a new process, prematurely breaking off the command string.

I guess, without looking at the source code though, that arguments to `mount` are passed as-is, and using PHP's `escapeshellarg` would fix this issue.

**Comment by kdubious** [comment]
Can we universally use `escapeshellarg` in our `sysCmd` function? Do we need to break out a new function for these cases that process user input?

**Comment by hondagx35** [comment]
no we can't use this in sysCmd generally.
We should escape all user inputs instead.
$auth = "username=".$mp['username'].",password=".escapeshellarg($mp['password']);

**Comment by ACXgit** [comment]
I agree with Frank, as it could introduce issues if used everywhere. An option could also be modifying the sysCmd function to accept it as parameter, and use it only where required.
